### PR TITLE
Fix/dal query 404 issue

### LINKED ISSUE #7637: tests: get_id_of_radius_audit_log_entry FAILURE
**Describe the problem**
```
            • get_id_of_radius_audit_log_entry FAILURE
    Testcase "get_id_of_radius_audit_log_entry", step #0: Assertion "result.statuscode ShouldEqual 200" failed. expected: 200  got: 404 (/usr/local/pf/t/venom/test_suites/wired_dot1x_eap_peap/55_check_radius_audit_log.yml:85)
    Testcase "get_id_of_radius_audit_log_entry", step #0: Assertion "result.bodyjson.items.items0 ShouldContainKey id" failed. unable to cast <nil> of type <nil> to map[string]interface{} (/usr/local/pf/t/venom/test_suites/wired_dot1x_eap_peap/55_check_radius_audit_log.yml:86)
```

**Mandatory informations**

 * Branch: Devel
 * Scenario: All scenarios that used it (fingerbank invalid db, eap peap ...)
 * Pipelines URL:
   * https://gitlab.com/inverse-inc/packetfence/-/jobs/4175026709 (NOTE: that pipeline is based on devel after the merge taht changed that part)
 
**Symptoms**
How the problem can be identified if it appears again ?

### PR DESCRIPTION
**Pull request by stgmsa**
# Description
(REQUIRED)
Moving db initialization into handler, added infinitive retry when establishing a db connection, 
added keep alive for existing db connection

# Impacts
(REQUIRED)
all go-dal related functions, and tests
CRUDs for logs tables admin_api_audit_log, auth_log, dns_audit_log, radacct_log, radius_audit_log, wrix. 

# Issue
The following syntax 'fixes #7637' will automatically closes a Github issue on pull-request merge time.
Modify the ISSUE_NUMBER in order to reflect the Github issue that need to be closed
fixes #7637 

# Delete branch after merge

YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

## Enhancements
* Enhancement 1
* Enhancement 2

## Bug Fixes
If an issue exists on Github, please refer to it (name) along with it's number...
* Issue with a query with 404 return of radius_audit_log/search API call in CI scenario. #7637

### CODE REVIEW

File: go/caddy/api/api.go [hunk 1]
```diff
@@ -58,12 +61,32 @@ func buildHandler(ctx context.Context) (APIHandler, error) {
...
-	NewAdminApiAuditLog().AddToRouter(router)
-	NewAuthLog().AddToRouter(router)
-	NewDnsAuditLog().AddToRouter(router)
-	NewRadacctLog().AddToRouter(router)
-	NewRadiusAuditLog().AddToRouter(router)
-	NewWrix().AddToRouter(router)
+	done := false
+	var DB *gorm.DB
+	var err error
+	for !done {
```
Comment (JeGoi) [inline go/caddy/api/api.go:67]: Shouldn't we limit that for a number of retry? like 2 or 3 max.
Then just exit and fail if not working

File: go/caddy/api/dal_wrix_test.go [hunk 2]
```diff
@@ -67,7 +70,10 @@ func removeDBTestEntriesWrix(t *testing.T, id string) error {
 
 func dalWrix() http.HandlerFunc {
 	router := httprouter.New()
-	NewWrix().AddToRouter(router)
+	ctx := context.Background()
+	dbs, _ := gorm.Open("mysql", db.ReturnURIFromConfig(ctx))
```
Comment (JeGoi) [inline go/caddy/api/dal_wrix_test.go]: Could we add a warning or something on the err in the logs?

File: go/caddy/api/dal_radius_audit_log_test.go [hunk 3]
```diff
@@ -67,7 +70,10 @@ func removeDBTestEntriesRadiusAuditLog(t *testing.T, id int64) error {
 
 func dalRadiusAuditLog() http.HandlerFunc {
 	router := httprouter.New()
-	NewRadiusAuditLog().AddToRouter(router)
+	ctx := context.Background()
+	dbs, _ := gorm.Open("mysql", db.ReturnURIFromConfig(ctx))
```
Comment (JeGoi) [inline go/caddy/api/dal_radius_audit_log_test.go]: Could we add a warning or something on the err in the logs?

File: go/caddy/api/dal_radacct_log_test.go [hunk 4]
```diff
@@ -68,7 +71,10 @@ func removeDBTestEntriesRadacctLog(t *testing.T, id int64) error {
 
 func dalRadacctLog() http.HandlerFunc {
 	router := httprouter.New()
-	NewRadacctLog().AddToRouter(router)
+	ctx := context.Background()
+	dbs, _ := gorm.Open("mysql", db.ReturnURIFromConfig(ctx))
```
Comment (JeGoi) [inline go/caddy/api/dal_radacct_log_test.go]: Could we add a warning or something on the err in the logs?

File: go/caddy/api/dal_dns_audit_log_test.go [hunk 5]
```diff
@@ -69,7 +72,10 @@ func removeDBTestEntriesDnsAuditLog(t *testing.T, id int64) error {
 
 func dalDnsAuditLog() http.HandlerFunc {
 	router := httprouter.New()
-	NewDnsAuditLog().AddToRouter(router)
+	ctx := context.Background()
+	dbs, _ := gorm.Open("mysql", db.ReturnURIFromConfig(ctx))
```
Comment (JeGoi) [inline go/caddy/api/dal_dns_audit_log_test.go]: Couldn't we add a warning or a trace in the logs if the er?

File: go/caddy/api/dal_auth_log_test.go [hunk 6]
```diff
@@ -68,7 +71,9 @@ func removeDBTestEntriesAuthLog(t *testing.T, id int64) error {
 
 func dalAuthLog() http.HandlerFunc {
 	router := httprouter.New()
-	NewAuthLog().AddToRouter(router)
+	ctx := context.Background()
+	dbs, _ := gorm.Open("mysql", db.ReturnURIFromConfig(ctx))
```
Comment (JeGoi) [inline go/caddy/api/dal_auth_log_test.go]: Could we add a warning or something on the err in the logs?

**Review by JeGoi — COMMENTED** [review]
I think it is always better to have more in logs than less in case of error, but it will not change the fact that the code should work like this.
Let's see the CI results: https://gitlab.com/inverse-inc/packetfence/-/pipelines/924351324

File: go/caddy/api/api.go (hunk 1 above)
Comment (stgmsa) [reply go/caddy/api/api.go:67]: We don't have to, since mysql is a must for most of the feature to be available.

**Pull request merged**